roles/create_mce_operator/tasks/main.yml

```diff
@@ -0,0 +1,184 @@
+            - enabled: true
+              name: cluster-lifecycle
+            - enabled: true
+              name: cluster-manager
+            - enabled: true
+              name: discovery
+            - enabled: true
+              name: hive
+            - enabled: true
+              name: server-foundation
+            - enabled: true
+              name: cluster-proxy-addon
+            - enabled: false
+              name: managedserviceaccount-preview
+            - enabled: false
+              name: hypershift-preview
+            - enabled: true
+              name: console-mce
+        targetNamespace: multicluster-engine
+
+- name: Wait until MCE Pods are running
+  kubernetes.core.k8s_info:
+    kind: Pod
+    namespace: multicluster-engine
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+  register: pod_list_2
+  vars:
+    filter_str: '[].{"name": metadata.name, "status": status.phase}'
+  until: pod_list_2.resources | json_query(filter_str) | rejectattr('name', 'in', old_pod_names) | map(attribute='status') | unique | list == ["Running"]
+  retries: 20
+  delay: 10
+
+- name: Patch the provisioning-configuration to enable CIM service
+  kubernetes.core.k8s:
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    state: present
+    api_version: metal3.io/v1alpha1
+    kind: provisioning
+    name: provisioning-configuration
+    definition:
+      spec:
+        watchAllNamespaces: true
+
+- name: Get MCE Pods list
+  kubernetes.core.k8s_info:
+    kind: Pod
+    namespace: multicluster-engine
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+  register: pod_list
+
+- ansible.builtin.set_fact:
+    old_pod_names: "{{ pod_list.resources | json_query('[].metadata.name')}}"
+
+- ansible.builtin.debug:
+    msg: "{{ old_pod_names }}"
+
+- name:  Create the AgentServiceConfig
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: agent-install.openshift.io/v1beta1
+      kind: AgentServiceConfig
+      metadata:
+       name: agent
+      spec:
+        databaseStorage:
+          storageClassName: ocs-storagecluster-cephfs
+          accessModes:
+          - ReadWriteOnce
+          resources:
+            requests:
+              storage: 20Gi
+        filesystemStorage:
+          storageClassName: ocs-storagecluster-cephfs
+          accessModes:
+          - ReadWriteOnce
+          resources:
+            requests:
+              storage: 10Gi
+        imageStorage:
+          storageClassName: ocs-storagecluster-cephfs
+          accessModes:
+          - ReadWriteOnce
+          resources:
+            requests:
+              storage: 20Gi
+        osImages:
+          - openshiftVersion: "{{ openshift_version }}"
+            version: "{{ version }}"
+            url: "{{ url }}"
+            rootFSUrl: "{{ rootfsurl }}"
+            cpuArchitecture: "x86_64"

```

Comment:
It seems like these should also be gotten from `assisted_installer_release_images` or from a variable provided by a user

Reply:
Actually, these variables are provided by user in my scenario. I have added these values in my inventory file.

Reply:
sorry I meant it should be `assisted_installer_os_images` 

By a variable I mean a structured variable like:
```
day_2_os_image: 
   openshiftVersion: "..."
   rhcos_version: "..."
   url: "..."
   rootFSUrl: "..."
   cpuArchitecture: "x86_64"
```

Having said that it would defiantly be preferable to use `assisted_installer_os_images` that way they should be pulled down my any other playbooks that pull down offline/mirror things for assisted installer. And there is only once place to go find os_images and releases.